internal/primitives/core/hash/hash.go

```diff
@@ -0,0 +1,54 @@
+package hash
+
+import (
+	"crypto/rand"
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// Fixed-size uninterpreted hash type with 32 bytes (256 bits) size.
```

Comment:
Why not just use out existing hash impl here? Does it just not have the needed receivers?

Reply:
Yea it doesn't have our existing receivers.  Also note the type of `string` here.  I need this cause I want it to satisfy `constraints.Ordered`.